### PR TITLE
add configurable custom callback URL.

### PR DESCRIPTION
**Pull request by michaelmcandrew**
Following discussion in https://github.com/cividesk/com.cividesk.email.sparkpost/issues/51, this PR adds a callback URL that can be configured.

It has been added as a setting (similar to other sparkpost settings).

When left blank (and it is blank by default) the extension uses the default callback URL. So the upgrade path should be smooth.

Let me know what you think.

Michael

### CODE REVIEW

File: settings/Sparkpost.setting.php [hunk 1]
```diff
@@ -17,6 +17,19 @@
     'description' => 'SparkPost REST API key',
     'help_text' => 'You can create API keys at: https://app.sparkpost.com/account/credentials',
   ),
+  'sparkpost_customCallbackUrl' => array(
+    'group_name' => CRM_Sparkpost::SPARKPOST_EXTENSION_SETTINGS,
+    'group' => 'com.cividesk.email.sparkpost',
+    'name' => 'sparkpost_customCallbackUrl',
+    'type' => 'String',
+    'html_type' => 'password',
```
Comment (nganivet) [inline settings/Sparkpost.setting.php]: Why 'password' here? We do not need to blank out what is entered for this setting.

Comment (michaelmcandrew) [reply settings/Sparkpost.setting.php]: oops, sorry, copy and paste error...

Comment (nganivet) [reply settings/Sparkpost.setting.php]: Please fully test the PR before I merge it. This was an obvious mistake. Thanks.

File: CRM/Admin/Form/Setting/Sparkpost.php [hunk 2]
```diff
@@ -41,6 +41,7 @@ public function buildQuickForm() {
     $this->add('password', 'sparkpost_apiKey', ts('API Key'), '', TRUE);
     $this->add('text', 'sparkpost_ipPool', ts('IP pool'));
     $this->addYesNo('sparkpost_useBackupMailer', ts('Use backup mailer'));
+    $this->add('text', 'sparkpost_customCallbackUrl', ts('Custom callback URL'));
```
Comment (michaelmcandrew) [inline CRM/Admin/Form/Setting/Sparkpost.php]: @nganivet: this is the line that specifies the form field type

File: settings/Sparkpost.setting.php (hunk 1 above)
Comment (michaelmcandrew) [reply settings/Sparkpost.setting.php]: @nganivet. I have tested it and it works nicely. FYI that was not an obvious mistake - that html_type setting type value is not surfaced in the extension. the significant field, type was correctly set as String.

**Comment by nganivet** [comment]
@michaelmcandrew Thanks, PR much appreciated.

**Comment by michaelmcandrew** [comment]
No problem - thanks for merging.

**Pull request merged**